cupy/linalg/_util.py

```diff
@@ -151,3 +151,19 @@ def _tril(x, k=0):
 def _triu(x, k=0):
     _triu_kernel(k, x)
     return x
+
+
+def batch_identity(batch_shape, n, dtype):
+    shape = batch_shape + (n, n)
+    idx = cupy.arange(n)
+    x = cupy.zeros(shape, dtype)
+    x[..., idx, idx] = 1
```

Comment:
The line can be `x.diagonal(axis1=-2, axis2=-1)[...] = 1` because CuPy doesn't make it readonly.